client/api/search.ts

```diff
@@ -76,6 +76,22 @@ export function searchRaw<T>(args: ISearchAPIParams): Promise<IRestApiResponse<T
     );
 }
 
+export const searchRawAndStore = <T extends Array<IEventOrPlanningItem>>(args: ISearchAPIParams) => (
+    (dispatch: any): Promise<IRestApiResponse<T>> => superdeskApi.dataApi.queryRawJson<IRestApiResponse<T>>(
+        'events_planning_search',
+        excludeNullParams(args)
+    ).then((res) => {
+        if (args.include_associated_planning) {
+            const relatedPlans = res._items.filter((item) => item.type === 'planning');
+
+            dispatch(planningApi.receivePlannings(relatedPlans));
```

Comment:
You could probably just import the `receivePlannings` function directly.
Don't forget to also perform
`dispatch(receiveEvents(events))`, so related events can also be loaded and stored